polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -176,32 +207,79 @@ pub enum ArtifactState {
 
 /// A container of all known artifact ids and their states.
 pub struct Artifacts {
-	artifacts: HashMap<ArtifactId, ArtifactState>,
+	inner: HashMap<ArtifactId, ArtifactState>,
 }
 
 impl Artifacts {
-	/// Initialize a blank cache at the given path. This will clear everything present at the
-	/// given path, to be populated over time.
-	///
-	/// The recognized artifacts will be filled in the table and unrecognized will be removed.
-	pub async fn new(cache_path: &Path) -> Self {
-		// First delete the entire cache. This includes artifacts and any leftover worker dirs (see
-		// [`WorkerDir`]). Nodes are long-running so this should populate shortly.
-		let _ = tokio::fs::remove_dir_all(cache_path).await;
+	#[cfg(test)]
+	pub(crate) fn new() -> Self {
+		Self { inner: HashMap::new() }
+	}
+
+	#[cfg(test)]
+	pub(crate) fn len(&self) -> usize {
+		self.inner.len()
+	}
+
+	/// Create a table with valid artifacts and prune the invalid ones.
+	pub async fn new_and_prune(cache_path: &Path) -> Self {
+		let mut artifacts = Self { inner: HashMap::new() };
+		Self::prune_and_insert(cache_path, &mut artifacts).await;
+		artifacts
+	}
+
+	// FIXME eagr: extremely janky, please comment on the appropriate way of setting
+	// * `last_time_needed` set as roughly around startup time
+	// * `prepare_stats` set as Nones, since the metadata was lost
```

Comment:
`last_time_needed` looks good to me.

`prepare_stats` may not even be necessary after the artifact has been prepared and the metrics logged. What happens when this field is removed?